README.md

```diff
@@ -1,32 +1,289 @@
+where the similarities end. Unlike a loop, iterator methods:
+
+1. are _called on_ an instance of an array, and
+2. take a _callback function_ as an argument.
+
+For `forEach`, the basic setup looks like this:
+
+```js
+<array>.forEach(callback);
+```
+
+For example, if we want to log the values of our array, we could do this:
+
+```js
+const myArray = [1, 2, 3];
+myArray.forEach(function(el) {
+    console.log(el);
+});
+```
+
+So what's happening here? We are calling `forEach()` on `myArray` and passing an
+anonymous function as the callback. The `forEach()` method iterates through
+`myArray` and, in each iteration, _automatically passes_ the current element to
+the callback function. (It actually automatically passes _three_ values, not
+just one; more about that later.) The callback captures the element using the
+`el` parameter then, in the body of the loop, logs it to the console. That's it!
+
+**Challenge:** there are other ways we could code this. We could call
+`forEach()` directly on the array rather than creating a variable, or we could
+use an arrow function for our anonymous function. Take a few minutes to see if
+you can code each of those before moving on. If you have difficulty with this
+task, don't worry too much. The important thing is to keep practicing!
+
+<details><summary><b>Calling `forEach()` directly on the array</b></summary><p>
+    <code>[1, 2, 3].forEach(function(el) {</code><br/>
+    <code>console.log(el);</code><br/>
+    <code>});</code>
+</p></details>
+
+<details><summary><b>Passing an arrow function as the callback</b></summary><p>
+    <code>[1, 2, 3].forEach((el) => console.log(el));</code>
+</p></details>
+
+Of course, we can also create the callback as a named function and pass it by
+name. That would look like this:
+
+```js
+function logElements(value) {
+    console.log(value);
+}
+
+[1, 2, 3].forEach(logElements);
+```
+
+Note that we don't need to explicitly pass the current element to the callback:
+the `forEach()` method does that for us! We do, however, need to set up our
+callback with a parameter, or maybe more than one...
+
+## More on `forEach()`
+
+We mentioned above that `forEach()` automatically passes _three_ values to our
+callback in each iteration, not just the current element. The other two values
+are the _index_ of the current element in the array and the array itself. We
+access them by setting up additional parameters in our callback. For example, we
+could log the contents of the array with a little more detail:
+
+```js
+const myArray = ["one", "two", "three"];
+
+function logArrayContents(element, index) {
+    console.log(`arr[${index}] = ${element}`);
+}
+myArray.forEach(logArrayContents);
+// => 
+// arr[0] = one
+// arr[1] = two
+// arr[2] = three
+```
+
+If we also want to use the array itself inside our callback, we would simply add
+a third parameter.
+
+**Note:** If we only use one parameter, the JavaScript engine will automatically
```

Comment:
Could be useful to summarize the three parameters with a small bullet point list as well, like 

1. First parameter: Array element 
2. Second parameter: etc. etc...